src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -362,46 +331,35 @@ internal PSKeyword GetLogKeywords()
             return result;
         }
 #endif // UNIX
-
-        private T ReadValueFromFile<T>(string fileName, string key)
+ 
+        private T ReadValueFromFile<T>(string fileName, string key, T defaultValue = default(T),
```

Comment:
Do we read option by option? I'd expect that we read whole config file, parse and apply.

Reply:
Yes, we currently read what is being requested, so that we don't need to parse unnecessary portion of the config file.

Reply:
I don't understand why we do that. We have some config files and registry - all so small that we'd read and parse them all in one step without performance issue.

Reply:
To me, it's an implementation detail to either read all settings at a time or allow the code to only read what it interests in. I don't favor one over another. The configuration APIs were already doing the latter, and I just follow that design.